messaging-client-databus/src/main/java/com/inmobi/messaging/consumer/databus/DatabusConsumer.java

```diff
@@ -88,6 +85,10 @@
   public static String clusterNamePrefix = "databusCluster";
   private Boolean readFromLocalStream;
   private int numList = 0;
+  private  List<String> collectorsWithReaders = new ArrayList<String>();
```

Comment:
@sreedishps @rajubairishetti  Actually, why do we even need these new lists -  collectorsWithReaders and startedReaders? 

AbstractMessagingDatabusConsumer. readers is a Map<PartitionId, PartitionReader>, where PartitionId contains clusterName and collector. So inside createPartitionReaders(), after getting the latest collectors for a cluster, we just need to create a PartitionId for each, check if they are already present in readers map. If not, simply add an entry there and start the new PartitionReader.

This way, we don't need any of - collectorsWithReaders, startedReaders, startNewReaders(). Also, if we can declare createPartitionReaders() as synchronized, we wouldn't even need initDone.